### PR TITLE
feat: add grpc lib to beagle

### PR DESCRIPTION
**Pull request by luismaximozup**
Adds a lib to support grpc in Beagle.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: grpc-backend-lib/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+kotlin.code.style=official
+protocVersion=3.15.0
+grpcVersion=1.35.0
+beagleVersion=1.7.0
```
Comment (LG95) [inline grpc-backend-lib/gradle.properties:4]: Why not have all dependency versions here?

File: grpc-backend-lib/src/main/kotlin/br/com/zup/grpc/BeagleGrpcUtil.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+private fun serializeNodeAttributes(jsonMap: MutableMap<String, Screen>): Map<String, Screen> {
+    val skip = listOf("_beagleComponent_", "context", "id", "style", "children", "child")
+    val attributes = HashMap<String, Screen>()
+    jsonMap.forEach {
+        if (!skip.contains(it.key)) attributes[it.key] = it.value
+    }
+    return attributes
+}
```
Comment (LG95) [inline grpc-backend-lib/src/main/kotlin/br/com/zup/grpc/BeagleGrpcUtil.kt:28]:
```suggestion
val skip = setOf("_beagleComponent_", "context", "id", "style", "children", "child")

fun serializeNodeAttributes(jsonMap: MutableMap<String, Screen>): Map<String, Screen> =
    jsonMap.filterKeys { !skip.contains(it) }
```

File: grpc-backend-lib/src/main/kotlin/br/com/zup/grpc/BeagleGrpcUtil.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+
+    return messageBuilder.build()
+}
+
+fun asGrpcView(widget: Screen): Messages.ViewNode {
+    val mapper = BeagleSerializationUtil.beagleObjectMapper()
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    val jsonMap = mapper.convertValue(widget, MutableMap::class.java) as MutableMap<String, Screen>
```
Comment (LG95) [inline grpc-backend-lib/src/main/kotlin/br/com/zup/grpc/BeagleGrpcUtil.kt:57]: Why is this map mutable and the value type `Screen`?

**Review by LG95 — COMMENTED** [review]
Why is gradle wrapper being partially deleted in this branch?

**Pull request merged**